### PR TITLE
Fixes #9025 -- Added deletion of WebAnalytics events in dataInsight Workflow

### LINKED ISSUE #9025: Add delete logic for webAnalytics Events
## Context
Web Analytics are ingested each time a user interacts with a web page. To limit the accumulation of events we need to delete events older than `n` days

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Fixes #9025.
- added DELETE endpoint for web analytics events
- added logic in data insight workflow to delete web analytics events older than 7 days

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/data_insight/api/workflow.py [hunk 1]
```diff
@@ -235,6 +237,17 @@ def _execute_kpi_runner(self):
...
 
+    def _execute_web_analytics_event_data_cleaning(self):
+        """We will delete web analytics events older than `RETENTION_DAYS`
+        to limit its accumulation
+        """
+        tmps = get_beginning_of_day_timestamp_mill(days=RETENTION_DAYS)
+        for web_analytic_event in WebAnalyticEventType:
+            self.metadata.delete_web_analytic_event_before_ts_exclusive(
+                web_analytic_event,
+                tmps,
```
Comment (nahuelverdugo) [inline ingestion/src/metadata/data_insight/api/workflow.py:248]: nit:
```suggestion
        tmsp = get_beginning_of_day_timestamp_mill(days=RETENTION_DAYS)
        for web_analytic_event in WebAnalyticEventType:
            self.metadata.delete_web_analytic_event_before_ts_exclusive(
                web_analytic_event,
                tmsp,
```
to align it with the name we use on `ingestion/src/metadata/ingestion/ometa/mixins/data_insight_mixin.py`.

File: ingestion/tests/integration/data_insight/test_web_analytic_events.py [hunk 2]
```diff
@@ -114,3 +114,53 @@ def test_web_analytic_events(self):
...
+
+        for delta in range(7):
+            tmps = int((datetime.utcnow() - timedelta(days=delta)).timestamp() * 1000)
+
+            user_id = uuid.uuid4()
+            session_id = uuid.uuid4()
+
+            event_data = WebAnalyticEventData(
+                eventId=None,
+                timestamp=tmps,
```
Comment (nahuelverdugo) [inline ingestion/tests/integration/data_insight/test_web_analytic_events.py:129]: nit:
```suggestion
            tmsp = int((datetime.utcnow() - timedelta(days=delta)).timestamp() * 1000)

            user_id = uuid.uuid4()
            session_id = uuid.uuid4()

            event_data = WebAnalyticEventData(
                eventId=None,
                timestamp=tmsp,
```

File: ingestion/tests/integration/data_insight/test_web_analytic_events.py [hunk 3]
```diff
@@ -114,3 +114,53 @@ def test_web_analytic_events(self):
         )
 
         assert test_event
+
+    def test_delete_web_analytic_event(self):
+        """Test web analytic event deletion"""
+
+        for delta in range(7):
+            tmps = int((datetime.utcnow() - timedelta(days=delta)).timestamp() * 1000)
```
Comment (nahuelverdugo) [inline ingestion/tests/integration/data_insight/test_web_analytic_events.py:122]: Can we call `get_beginning_of_day_timestamp_mill` here?

**Review by nahuelverdugo — APPROVED** [review]

File: ingestion/tests/integration/data_insight/test_web_analytic_events.py (hunk 3 above)
Comment (TeddyCr) [reply ingestion/tests/integration/data_insight/test_web_analytic_events.py:122]: Yep, we should be able to.

**Review by nahuelverdugo — DISMISSED** [review]

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**